Sources/TuistGenerator/Graph/GraphNode.swift

```diff
@@ -150,6 +157,42 @@ enum PrecompiledNodeError: FatalError, Equatable {
 }
 
 class SDKNode: GraphNode {
+    let name: String
+    let status: SDKStatus
+    let type: Type
+
+    init(name: String,
+         platform: Platform,
+         status: SDKStatus) throws {
+        let sdk = AbsolutePath("/\(name)")
+
+        guard let sdkExtension = sdk.extension,
```

Comment:
Instead of throwing here, I'd add a new linting rule that verifies that the extension is supported. That way developers can see all the validation issues at once and fix them before running `tuist generate` again.

Reply:
That means then that this `Error.unsupported` should be reported as a bug because we should never reach this point if the extension is not supported.

Reply:
Moving this to a lint rule would be much better - I did attempt that in the previous PR, sadly was unable to as the graph is created before any linting is done.

One idea is to always construct an `SDKNode` even if an invalid name is passed in (e.g. `Foo` without the extension, or `Foo.bar` not a `.framework` and `.tbd`) and only work out the paths and type later in the pipeline - that way we're able to lint the names. Can play around with this see if it works.

Another is possibly to be more explicit to avoid user error in the first place (will be a breaking change - but maybe ok if we feel its a better choice). 

```swift
dependencies: [
   .sdk(name: "CloudKit", type: .framework),
   .sdk(name: "ARKit", type: .framework, status: .optional),
   .sdk(name: "libc++", type: .tbd),
]
```

Reply:
> Another is possibly to be more explicit to avoid user error in the first place (will be a breaking change - but maybe ok if we feel its a better choice).

I like this one :+1:, even though it's a breaking change.

Reply:
Would you be ok for that being a follow up PR to reduce the scope of changes in this one?

Reply:
Yes!

Reply:
Added #422